File: SpringBoot-SecKill-Scala/src/main/scala/cn/edu/jxnu/seckill/redis/key/AccessKey.scala
package cn.edu.jxnu.seckill.redis.key

import cn.edu.jxnu.seckill.redis.BasePrefix

/**
 * 定义Redis 访问键 前缀
 *
 * @author 梦境迷离.
 * @time 2018年5月20日
 * @version v1.0
 */
class AccessKey private (expireSe: Integer, prefix: String)
    extends BasePrefix(expireSe, prefix) {

}

object AccessKey {

    def withExpire(expireSeconds: Integer): AccessKey = {
        new AccessKey(expireSeconds, "access")
    }
}

File: seckill/src/main/java/cn/edu/jxnu/seckill/redis/AccessKey.java
package cn.edu.jxnu.seckill.redis;

import cn.edu.jxnu.seckill.redis.key.BasePrefix;

public class AccessKey extends BasePrefix {
	private AccessKey(int expireSeconds, String prefix) {
		super(expireSeconds, prefix);
	}

	public static AccessKey access = new AccessKey(5, "access");

	public static AccessKey withExpire(int expireSeconds) {
		return new AccessKey(expireSeconds, "access");
	}
}
